### PR TITLE
feat!(share): adding config variables for blocksync

### LINKED ISSUE #1465: node/share: Data sync modes for FN
## Context

We need a configuration to run an FN in reconstruction-only mode, so we can continue benchmarking reconstruction performance and apply various obvious optimizations coming from __happy path__ integration.

This should be done by configuring the node to run without `ShrExGetter`

### PR DESCRIPTION
**Pull request by distractedm1nd**
## Overview
Closes #1465 
Based on #1729

### CODE REVIEW

File: share/getters/shrex.go [hunk 1]
```diff
@@ -28,6 +28,23 @@ type ShrexGetter struct {
...
+	}
+}
+
+func (sg *ShrexGetter) Start(ctx context.Context) error {
+	return sg.peerManager.Start(ctx)
+}
+
+func (sg *ShrexGetter) Stop(ctx context.Context) error {
+	return sg.peerManager.Stop(ctx)
+}
```
Comment (Wondertan) [inline share/getters/shrex.go:46]: If ShrexGetter maintains the lifecycle of `peer.Manager`, it should also construct it, i.e. `peer.Manager` is a child of ShrexGetter but not a dependency. This also means that peer manager won't be part of FX dep graph, which is totally ok as the peer manager is a single purpose.

I don't wanna push this, and we can do it during later refactorings, so can make an issue

**Review by Wondertan — COMMENTED** [review]
Have only one requests

File: share/getters/shrex.go (hunk 1 above)
Comment (Wondertan) [reply share/getters/shrex.go:46]: Consider this is a feedback for https://github.com/celestiaorg/celestia-node/pull/1729

File: nodebuilder/share/constructors.go [hunk 2]
```diff
@@ -57,3 +65,30 @@ func ensureEmptyCARExists(ctx context.Context, store *eds.Store) error {
...
+	shrexSub *shrexsub.PubSub,
+	discovery *disc.Discovery,
+	host host.Host,
+	connGater *conngater.BasicConnectionGater,
+) *peers.Manager {
+	// TODO: find better syncTimeout duration?
+	return peers.NewManager(headerSub, shrexSub, discovery, host, connGater, modp2p.BlockTime*5)
+}
+
+func fullGetter(
```
Comment (renaynay) [inline nodebuilder/share/constructors.go:80]:
```suggestion
func cascadeGetter(
```

**Review by renaynay — COMMENTED** [review]
looks like this contains commits from other pr

File: nodebuilder/share/config.go [hunk 3]
```diff
@@ -18,19 +18,27 @@ type Config struct {
 	// AdvertiseInterval is a interval between advertising sessions.
 	// NOTE: only full and bridge can advertise themselves.
 	AdvertiseInterval time.Duration
+	// GetterTimeout is the duration that each share retrieval getter will be run before moving to
+	// the next getter.
+	GetterTimeout time.Duration
+	// UseShareExchange is a flag toggling the usage of shrex protocols for blocksync.
+	// NOTE: This config variable only has an effect on full and bridge nodes.
+	UseShareExchange bool
```
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply nodebuilder/share/config.go:26]: No, it is a tracked issue, the one this pr is trying to close #1465

File: nodebuilder/share/constructors.go (hunk 2 above)
Comment (distractedm1nd) [reply nodebuilder/share/constructors.go:80]: light nodes will also use a cascadeGetter

File: nodebuilder/share/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Wondertan) [reply nodebuilder/share/config.go:26]: For testing, yes

File: nodebuilder/share/constructors.go (hunk 2 above)
Comment (Wondertan) [reply nodebuilder/share/constructors.go:80]: light nodes will also use a cascadeGetter

**Review by Wondertan — APPROVED** [review]

File: nodebuilder/share/constructors.go (hunk 2 above)
Comment (distractedm1nd) [reply nodebuilder/share/constructors.go:80]: light nodes will also use a cascadeGetter

Comment (renaynay) [reply nodebuilder/share/constructors.go:63]: light nodes will also use a cascadeGetter

**Review by renaynay — APPROVED** [review]
have u tested these default params yet?

**Pull request merged**